sherpa/astro/data.py

```diff
@@ -2676,32 +2680,18 @@ def _group_to_channel(self, val, group=True, response_id=None):
         # Convert to an integer (this keeps the channel within
         # the group).
         #
-        mid = np.floor(mid)
-        val = np.asarray(val).astype(np.int_) - 1
-        try:
-            return mid[val]
-        except IndexError:
-            raise DataErr(f'invalid group number: {val}') from None
+        return np.floor(mid)
 
-    def _channel_to_energy(self, val, group=True, response_id=None):
+    def _channel_to_energy(self, group=True, response_id=None):
         elo, ehi = self._get_ebins(response_id=response_id, group=group)
-        val = np.asarray(val).astype(np.int_) - 1
-        try:
-            return (elo[val] + ehi[val]) / 2.0
-        except IndexError:
-            raise DataErr('invalidchannel', val) from None
+        return (elo + ehi) / 2.0
```

Comment:
I'm not a big fan on functions that are just one or two lines long. This is (according to the comment added above) only use in one spot. I would thus suggest to inline all three of those functions into `_channel_to_energy`. The difference in runtime is negligible, but the difference in readability is not.

Reply:
I don't understand what you mean here. If you look at `def _set_units` you see we basically have

```
if using channels:
    self._from_channel = self._group_to_channel
elif using energy:
    self._from_channel = self._channel_to_energy
elif using wavelength:
    self._from_channel = self._channel_to_wavelength
else
    oops, this should not be possible
```

So it's not clear to me how you would want to inline them.

There is the possibility of doing a more-serious rework of this code, but I do not see the benefit of this **at this time** as I'm trying to focus of other parts of the DataPHA class.

Reply:
Ok, let's worry about that later.